### DISCUSSION TITLE
v3.1.0: Invalid flag: --release when compiling common

### DISCUSSION THREAD

**Discussion by rickfish** [Netflix/conductor#2414] [Q&A] [answered]
Building Conductor v3.1.0 on my local home machine works fine but when building at my work location we are getting an InvalidArgumentException when building the common module: 

`invalid flag: --release`

At my work location we cannot go out to the Internet to get packages so I know that is the cause but I cannot figure out what is the source of the error. We have an internal artifactory repo that houses our packages so all the 'repositories' sections in our build.gradle files have to point to that artifactory instance.

I know that this is a result of setting options.release=8 in the main build.gradle file for the common module but I cannot figure out how to fix it. I am using AdoptOpenJDK v11.0.6 as the JDK in my path and that is what JAVA_HOME is pointing to as well. All of my google searches have solved the same error when using maven or IntelliJ IDEA and they say that the metadata files for those tools point to the wrong JDK but this is building from the command line of GitBash using gradlew.

Any help would be much appreciated if anyone is facing this same issue.

**Comment by guru1306** [comment]
I faced this issue today when JAVA 8 was installed on my machine. Once I installed Open JDK JAVA 11 the issue got fixed. My machine is ubuntu. I don't have JAVA_HOME set

![image](https://user-images.githubusercontent.com/12997511/129932508-86c9d7ae-a90a-44a6-96b4-0bf164907bfc.png)

Are you using gradle or gradlew? Looks like slightly older Gradle versions had an issue with AdoptOpenJDK. gradlew downloads the version mentioned in the gradle properties file

https://github.com/AdoptOpenJDK/homebrew-openjdk/issues/55

**Comment by rickfish** [comment, marked as answer]
@guru1306 thanks...i am using gradlew which is using 6.8.3. I upgraded to your version ov openjdk (11.0.11) and am still having the same problem, I even removed JAVA_HOME. There must be something else to do with my environment or it is GitBash vs Ubuntu...
![conductor-error](https://user-images.githubusercontent.com/39354200/129963861-5df19ba3-d941-452a-bcd4-be4859aabc50.png)

**Reply by guru1306** [reply]
It looks like you have still not uninstalled all AdoptOpenJDK versions.

**Reply by rickfish** [reply]
@guru1306 you are right, still had the other version installed...didn't think it made a diff since I had the latest one in my PATH. I just changed my gradle.properties to point to 11.0.11 and it worked (unfortunately I didn't follow your last comment about gradle.properties). Thanks!

**Comment by sumi266** [comment]
Trying to build and run conductor instance through intellij but getting the error :

*** What went wrong:
Execution failed for task ':conductor-common:compileJava'.
> invalid flag: --release**

I have my module sdk pointing to JDK 11 .

**Reply by rickfish** [reply]
@sumi266 what is your org.gradle.java.home set to in your gradle.properties file? Once I set that to my JDK 11 folder it fixed the problem...

**Reply by sumi266** [reply]
@rickfish  thanks a ton . Was able to resolve the problem by pointing  to JDK 11.
